### PR TITLE
feat: automated integration testing process

### LINKED ISSUE #103: Config Docker de Testes de Integração
**Describe the feature**

Esta funcionalidade propõe a criação de um setup de Docker Compose para testes de integração em ambientes de projeto Go, visando facilitar a execução de testes em condições semelhantes às de produção de forma isolada e reproduzível.

<details open>
  <summary>
    <b>Description</b>
  </summary>
   Esta solicitação de funcionalidade envolve a configuração e criação de um setup de Docker Compose para execução de testes de integração em um ambiente de projeto Go (Golang). O objetivo é otimizar o processo de testes permitindo que os desenvolvedores facilmente configurem ambientes de teste isolados que imitam de perto as configurações de produção, assegurando que os testes de integração sejam executados de maneira consistente e controlada através de diferentes configurações de desenvolvimento.
</details>

<br>

<details> 
  <summary>
    <b>Use Case</b>
  </summary>
  
  Os testes de integração desempenham um papel crucial no ciclo de vida do desenvolvimento, garantindo que os componentes individuais de um sistema de software funcionem juntos conforme esperado. No contexto de projetos Go, essa funcionalidade beneficia os desenvolvedores fornecendo um ambiente de teste reproduzível e isolado, utilizando contêineres Docker. Essa configuração é particularmente útil para testar interações com serviços externos ou bancos de dados, onde simular condições semelhantes à produção é essencial para resultados de testes precisos. Ela melhora a confiabilidade dos testes, facilita os processos de integração contínua (CI) e suporta práticas de DevOps, permitindo que os desenvolvedores testem alterações localmente em uma configuração que espelha o ambiente de CI, reduzindo o problema de "funciona na minha máquina".
</details>

<br>

<details> 
  <summary>
    <b>Implementation Details</b>
  </summary>

Para implementar essa funcionalidade, um arquivo docker-compose.yml será criado na raiz do projeto Go. Este arquivo definirá os serviços, redes e volumes necessários para os testes de integração. Componentes chave incluirão:

Serviço Go: Define o contêiner da aplicação, construído a partir de um Dockerfile que configura o ambiente Go e inclui o código do projeto.
Serviços Externos: Contêineres para quaisquer serviços externos com os quais a aplicação interage, como bancos de dados, filas de mensagens ou APIs de terceiros, instâncias simuladas ou reais conforme necessário.
Redes: Redes personalizadas para facilitar a comunicação segura e eficiente entre o contêiner da aplicação Go e os serviços externos.
Volumes: Volumes persistentes para serviços que requerem armazenamento de dados, garantindo que os dados sejam mantidos entre execuções de testes, se necessário.
A configuração do Docker Compose será projetada para ser invocada com um comando simples, como docker-compose up -d --build, para executar os testes de integração, e docker-compose down para desmontar o ambiente após os testes. Adicionalmente, um guia ou script pode ser fornecido para simplificar essas operações para os desenvolvedores.

</details>

<br>

<details> 
  <summary>
    <b>Visual Concepts</b>
  </summary>

Nenhum conceito visual específico é aplicável para esta funcionalidade🐳🐳

</details>

<br>

<details> 
  <summary>
    <b>Additional Information</b>
  </summary>
  

</details>

### PR DESCRIPTION
**Pull request by itsfpbtw**
Closes 
#103 

<details open> 
  <summary>
    <b>Feature</b>
  </summary>

Services and a dedicated workflow for integration testing using docker compose.
</details>

<details open> 
  <summary>
    <b>Bugfix</b>
  </summary>

- **Description**
N/A

- **Cause**
N/A

- **Solution**
N/A
</details>

<details> 
  <summary>
    <b>Changelog</b>
  </summary>
N/A
</details>

<details open> 
  <summary>
    <b>Visual evidences :framed_picture:</b>
  </summary>

</details>

<details open> 
  <summary>
    <b>Checklist</b>
  </summary>

  - [x] Issue linked
  - [ ] Build working correctly
  - [ ] Tests created
</details>

<details> 
  <summary>
    <b>Additional info</b>
  </summary>
N/A
</details>

### CODE REVIEW

File: .env.example
Comment (k1nha) [inline .env.example:1]: porque a mudamos a env pra lowercase?

File: api/routes/router.go
Comment (k1nha) [inline api/routes/router.go:1]: e se for mudar. temos que validar isso aqui também 
`!= development`

**Review by k1nha — APPROVED** [review]

**Comment by PiluVitu** [comment]
Pr do splash era para estar listado nos commits dessa branch ?

File: api/routes/routes.go [hunk 1]
```diff
@@ -2,22 +2,20 @@ package routes
...
-	FindPetController *petcontroller.FindPetController
 	UserController    *controllers.UserController
 }
 
 func InitRoutes(controllers Controllers, c *chi.Mux) {
 
 	c.Route("/api", func(r chi.Router) {
 		r.Route("/pets", func(r chi.Router) {
-			r.Get("/{id}", controllers.FindPetController.FindPet)
+			r.Get("/{id}", controllers.PetController.FindPet)
```
Comment (devlefel) [inline api/routes/routes.go:18]: modifica para petID para deixar na mesma logica

File: infra/db/pet_repository.go
Comment (devlefel) [inline infra/db/pet_repository.go:1]: Galera, vamos usar o sqlx  como padrao?
https://github.com/jmoiron/sqlx

Ja usamos em Usuario

**Review by devlefel — CHANGES_REQUESTED** [review]

**Comment by PiluVitu** [comment]
Esse é o atual ambiente de desenvolvimento, vou jogar as imagens para alpine, ta batendo um tamanho considerável nas imagens
image

**Comment by PiluVitu** [comment]
Preciso: 
- Ajustar workflow de teste unitario  
- Rodar as migrations no build

**Pull request closed without merging**